src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -496,51 +801,316 @@ private async Task<ReviewListItemModel> ToggleReviewApproval(ClaimsPrincipal use
         /// <returns></returns>
         private async Task UpdateReviewsUsingPipeline(string language, LanguageService languageService, int backgroundBatchProcessCount)
         {
-            var reviews = await _reviewsRepository.GetReviewsAsync(language: language, isClosed: false);
-            var paramList = new List<APIRevisionGenerationPipelineParamModel>();
-
-            foreach (var review in reviews)
+            try
             {
-                var revisions = await _apiRevisionsManager.GetAPIRevisionsAsync(review.Id);
-                foreach (var revision in revisions)
+                // Get all non-closed reviews for the language
+                var reviews = await _reviewsRepository.GetReviewsAsync(language: language, isClosed: false);
+                
+                var eligibleReviews = new List<APIRevisionGenerationPipelineParamModel>();
+                
+                foreach (var review in reviews)
                 {
-                    foreach (var file in revision.Files)
+                    try
                     {
-                        //Don't include current revision if file is not required to be updated.
-                        // E.g. json token file is uploaded for a language, specific revision was already upgraded.
-                        if (!file.HasOriginal || file.FileName == null || !languageService.IsSupportedFile(file.FileName) || !languageService.CanUpdate(file.VersionString))
+                        var revisions = await _apiRevisionsManager.GetAPIRevisionsAsync(review.Id);
+                        // Find the first revision that has original files and can be updated
+                        APIRevisionListItemModel firstRevision = revisions.FirstOrDefault(r => 
+                            r.Files.Any(f => f.HasOriginal) && 
```

Comment:
Why is this change as part of namespace approval feature? How is this related to namespace approval? You are also missing a condition here. `!languageService.IsSupportedFile(file.FileName)`